add workaround/option to select Sensu config file from within tool

bparli opened the pull request:
I am using your sensu-client-go package to extend one of our tools to also run as a standalone sensu check.  Since the tool already has command line flags I wasn't able to set the Sensu configuration file from the command line as the NewConfigFromFlagSet function expects.

- Return error in NewConfigFromFlagSet if flagset.configFile is not set
- Add exported function to set the Sensu config file from within the context of the tool to allow for setting the Sensu config file location

Commit:
add workaround/option to select Sensu config file from within tool

mihaitodor commented on sensu/config.go:
```diff
@@ -61,9 +76,10 @@ func NewConfigFromFlagSet(flagset *configFlagSet) (*Config, error) {
 		if err := json.Unmarshal(buf, &cfg.config); err != nil {
 			return nil, err
 		}
+		return &cfg, nil
```
Would it be possible to replace the logic inside this `if` block with a call to `NewConfigFromFile` in order to avoid code duplication?

mihaitodor commented on sensu/config.go:
```diff
@@ -61,9 +76,10 @@ func NewConfigFromFlagSet(flagset *configFlagSet) (*Config, error) {
 		}
+		return &cfg, nil
 	}
+	return nil, errors.New("No config file is set")
```
I think we don't want to return an error here, because the client is able to subscribe to checks from the Sensu server via the `SENSU_CLIENT_SUBSCRIPTIONS` environment variable and you can use `RABBITMQ_URI` to specify the URI of the Sensu server, which I believe lets you run the client without a config file.

AlexisMontagne replied:
Exactly, internally we only use the environment variables to configure the clients. It's a desired behavour.

Commit:
clean up duplicate code and remove error

bparli commented:
I've cleaned up the duplicate code and also removed the error return to realign with your conventions.

You've obviously thought about this more than me, but it seems to me using Environment variables like this deviates from the main sensu project conventions.  More importantly though, from what I can tell you could hose the sensu server if the SENSU_CLIENT_NAME environment variable is fat-fingered or not set. 

In any case, thanks for the PR review

AlexisMontagne commented on sensu/config.go:
```diff
@@ -48,22 +48,32 @@ func split(str string, token string) []string {
-		if err := json.Unmarshal(buf, &cfg.config); err != nil {
+func NewConfigFromFlagSet(flagset *configFlagSet) (*Config, error) {
+	if flagset != nil && flagset.configFile != "" {
+		cfg, err := NewConfigFromFile(flagset, flagset.configFile)
```
you can just do: `return NewConfigFromFile(flagset, flagset.configFile)`

bparli replied:
You're right, thats much better

Commit:
replace redundant error with returning tuple in NewConfigFromFlagSet

AlexisMontagne commented on sensu/config.go:
```diff
@@ -48,22 +48,32 @@ func split(str string, token string) []string {
+		return cfg, nil
+	} else {
+		var cfg = Config{flagset, &configPayload{}}
+		return &cfg, nil
```
you don't need to assign a variable, you can directlydo `return &Config{flagset, &configPayload{}}, nil`

AlexisMontagne commented on sensu/config.go:
```diff
@@ -48,22 +48,32 @@ func split(str string, token string) []string {
 			return nil, err
 		}
+		return cfg, nil
+	} else {
```
this `else` is not really needed, you can just drop it

AlexisMontagne reviewed (commented):
@bparli i gave you some syntax feedbacks, you can use the [golang linter](https://github.com/golang/lint) and [govet](https://golang.org/cmd/vet/) to catch these smells easily

Commit:
clean up return and remove else statement

AlexisMontagne reviewed (approved):
LGTM
